src/Microsoft.Diagnostics.Runtime/ModuleInfo.cs

```diff
@@ -82,6 +82,24 @@ public abstract class ModuleInfo
             return result;
         }
 
+        /// <summary>
+        /// Attempts to create a root resource node for the PE module.
+        /// </summary>
+        /// <param name="reader">The data reader to create this module from.</param>
+        /// <param name="imageBase">The base address of this module.</param>
+        /// <param name="imageSize">The module image size.</param>
+        /// <param name="isFileLayout">if true, the PE has file layout, false loaded layout</param>
+        /// <returns>IResourceNode instance</returns>
+        public static IResourceNode? TryCreateResourceRoot(IDataReader reader, ulong imageBase, ulong imageSize, bool isFileLayout)
```

Comment:
Does this need to be static?  Or should it be instance?  Are you intending to use this to read resources on things that aren't strictly wrapped by ModuleInfo?

Reply:
I want it to be static because I want to read resources from modules that don't have a ModuleInfo (like when SOS infra is hosted under windbg or lldb).

Reply:
Here is an example of how I'm going to use it: https://github.com/mikem8361/diagnostics/blob/42b6643ed799c1259aedd023c74380b2c5fa3abc/src/Microsoft.Diagnostics.DebugServices.Implementation/DataReader.cs#L208

Reply:
Sounds good.